### ISSUE TITLE
Limit the number of multi Image uploads

### ISSUE THREAD

**Issue by mikemastercorp** [thetbw/jquery-uploader#1]
Hello. I love your jquery uploader and I would be glad to include it in my Laravel Project.

In order for me to use it in the project, I would like to complete 2 things:

1. Limit the number of uploaded images for the multi type (I've checked the code but did not find any parameter/option to allow that).
2. Strip out the uploading functionality as I intend on using the built in Laravel abilities to update DB and store files. Are there any instructions which functions are used for adding the images to the field and the preview/deletion block?

In other words, I am trying to use the field for administering a multi image upload form but I would use your functionality to only preview/remove images and pass the array to Laravel?

**Comment by thetbw** [comment]
first one, I haven't written the file type check yet, but you can do it like this:

```javascript
$("#demo2").on("before-upload",(event,file)=>{
        console.log("file",file)
        //check file type
        // ....

        //call remove the file 
        $("#demo2").uploader("removeFiles",file.id)
        
        //show dialog or other
        // ...

        //finally ,throw a exception to break upload
        throw "not allow type"
 })
```

this  file object like this:
![image](https://user-images.githubusercontent.com/31307256/209051103-c5679068-0b53-414b-9663-c3fa58866332.png)

> notice : the type field just have `image` and `other` enmu ,you can check it by origin `file` field

**Comment by thetbw** [comment]
second：
you can set default value like this
```javascript
$("#demo3").uploader({
        autoUpload:false,
        multiple: true,
        defaultValue: [
            {
                name: "defa",
                url: "https://s2.loli.net/2022/05/07/GXrnozOj86pbFyl.jpg"
            }, {
                name: "minecreaft",
                url: "https://www.minecraft.net/etc.clientlibs/minecraft/clientlibs/main/resources/img/GrassBlock_HighRes.png"
            }
        ]
    })
```
autoConfig set false to close upload

default value obj can be a string separated by commas, the string is file remote link .
or  a file obj array.

call `uploadFiles` method to get all current file array .
You can edit this data in any way，and call `refreshPreviewFileList` to refresh component.
``` javascript
let files = $("#demo3").uploader("uploadFiles")

//edit the file array,add a obj or delete, other wise 
//...

//refresh
$("#demo3").uploader("refreshPreviewFileList")
```

this file obj  structure
```
{
    id:'afcaa3a2-028e-4556-8a83-e019141f4d6f', //file uuid,auto generate
    name: 'a.png', //file name
    /** 
    *   status:
    *    selected,uploading,uploaded,error,initial
    * /
    status: 'selected' ,
    type:'image' // image or other
    // url is the file preview url 
    url: 'blob:http://localhost:63342/8b5d7d5f-35df-4c4a-8738-59c00fe922af',
    file: Object,//this js file object,origin file,
    $ele: Object // this preview card dom, you better not create it manually
}
```
if you want create a file obj,
call `handleFileAdd` method, parameter is a origin file obj  array.

finally ,you can see this method code in this file https://github.com/thetbw/jquery-uploader/blob/master/src/uploader.js
